README.rst

```diff
@@ -5,6 +5,62 @@ This repository contains kiwi_ definitions to build containers for
 `SUSE OpenStack Cloud`_.
 The containers are built in the `Open Build Service`_.
 
+Generating the .kiwi files from templates
+=========================================
+
+We want to build containers based on different base containers (eg. for
+openSUSE or SLE). To accomplish that, we use Jinja2 templates and
+YAML based (.yml) files which contain variables to fill the templates.
+
+The tool to combine the single template (usually .kiwi.j2) and the
+variable files(s) (usually .yml) is `j2gen`_
+
+`j2gen` allows multiple variable files which are combined together (for
+dict/hash like data structures, the structures are merged where the last
+occurance wins).
+So we can have a chain of variable files like:
+
+- common/global.yml (used for *all* containers)
+- $image/common.yml (used for all $image containers)
+- $image/$distro.yml (used for the $distro specific container)
+- $image/$image.kiwi.j2 (the template to build $image for all distros)
+
+Theses are combined (eg. for a memcached SLE15SP1 image) with::
```

Comment:
"The .kiwi output file (eg. for a memcached SLE15SP1 image) can be generated from the above files with::" ?

Reply:
Done